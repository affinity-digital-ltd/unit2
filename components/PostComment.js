import { Component } from 'react'

import CommentForm from '../components/CommentForm'

import Auth from '../lib/Auth'

class PostComment extends Component {
  constructor (props) {
    super(props)

    this.auth = new Auth()
    this.login = this.login.bind(this)
  }

  login () {
    this.auth.login()
  }

  render () {
    if (this.auth.isAuthenticated()) {
      return <CommentForm postComment={this.props.mutation} />
    } else {
      return <div className='card'>
        <div className='card-body'>
          <p>If you would like to leave a comment, then please login</p>
          <a className='btn btn-success btn-lg' onClick={this.login} >Login</a>
        </div>
      </div>
    }
  }
}

export default PostComment
