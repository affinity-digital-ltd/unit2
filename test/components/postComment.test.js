import PostComment from '../../components/PostComment'
jest.mock('../../lib/Auth')

describe('<PostComment />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <PostComment />
    )

    expect(component).toMatchSnapshot()
  })
})
