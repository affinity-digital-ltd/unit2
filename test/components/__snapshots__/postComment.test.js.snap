// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostComment /> it renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <p>
      If you would like to leave a comment, then please login
    </p>
    <a
      className="btn btn-success btn-lg"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
</div>
`;
